ebuild-writing/bundled-deps/text.xml

```diff
@@ -0,0 +1,390 @@
+for whatever reason are opposed to these changes, the developers of
+<e>P</e> may want to fork <e>D</e>.
+</p>
+
+<p>
+But publishing and properly maintaining a fork takes time and effort, so
+the developers of <e>P</e> could be tempted to take the easy road, bundle
+their patched version of <e>D</e> with <e>P</e>, and maybe occasionally
+update it for upstream <e>D</e> changes.
+</p>
+</body>
+</subsection>
+</section>
+
+<section>
+<title>Problems</title>
+<body>
+
+<p>
+So why is bundling dependencies and static linking bad after all?
+</p>
+</body>
+
+<subsection>
+<title>Security implications</title>
+<body>
+
+<p>
+Let's consider you're a developer of <e>foo</e> and your <e>foo</e> uses
+<e>libbar</e>.
+</p>
+
+<p>
+Now, a very important security flaw has been found in <e>libbar</e>
+(say, remote privilege escalation). The problem is large enough that devs
+of <e>libbar</e> release fixed version right away, and distributions package
+it quickly to decrease the possibility of break-in to users' systems to
+minimum.
+</p>
+
+<p>
+If a particular distribution has efficient security upgrade system, the
+patched library can get there in less than 24 hours. But that would be of
+no use to <e>foo</e> users which will still use the earlier vulnerable library.
+</p>
+
+<p>
+Now, depending on how bad things are:
+</p>
+
+<ul>
+  <li>
+    If <e>foo</e> statically linked against <e>libbar</e>, then the users would
+    either have to rebuild <e>foo</e> themselves to make it use the fixed library
+    or distribution developers would have to make a new package for <e>foo</e> and
+    make sure it gets to user systems along with <e>libbar</e> (assuming they
+    are aware that the package is statically linked)
+  </li>
+  <li>
+    If <e>foo</e> bundled local copy of <e>libbar</e>, then they would have to wait
+    till you discover the vulnerability, update <e>libbar</e> sources, release
+    the new version and distributions package the new version.
+  </li>
+</ul>
+
+<p>
+In the meantime, users probably even won't know they are running vulnerable
+application just because they won't know there's a vulnerable library
+statically linked into the executables.
+</p>
+
+<p>
+Examples:
+</p>
+
+<ul>
+  <li>
+    <uri link="https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-3074">
+    CVE-2016-3074</uri> has to be
+    <uri link="https://bugs.php.net/bug.php?id=71912">fixed in PHP</uri>
+    (where it is bundled) after it is
+    <uri link="https://github.com/libgd/libgd/commit/2bb97f407c1145c850416a3bfbcc8cf124e68a19">
+    fixed in libgd</uri> (upstream).
+  </li>
+</ul>
+</body>
+</subsection>
+
+<subsection>
+<title>Waste of hardware resources</title>
+<body>
+
+<p>
+Say a media player is bundling library libvorbis. If libvorbis is also
+installed system-wide this means that two copies of libvorbis# occupy twice
+as much space on disk # occupy (up to) twice as much RAM (of the page cache)
```

Comment:
What are these stray `#` in the text? Leftover from a numbered list in RestructuredText or Markdown?

Reply:
Ah, it's numbered list markup from mediawiki...